UnityProject/Assets/Scripts/Objects/Other/DropPod.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Collections;
+using UnityEngine;
+using Mirror;
+using Systems.Explosions;
+
+namespace Objects
+{
+	/// <summary>
+	/// Spawns a drop pod. Falls from the "sky"!
+	/// </summary>
+	public class DropPod : NetworkBehaviour
+	{
+		private const int DROP_HEIGHT = 32; // How many tiles high the drop pod should spawn from the landing zone.
+		private const int TRAVEL_TIME = 4; // How many seconds it takes to get from spawn point to landing zone.
+		private const int EXPLOSION_STRENGTH = 80; // The strength of the explosion upon landing. 80 ~= soft crit.
+
+		[Tooltip("Whether the drop-pod should spawn stationary or falling.")]
+		[SerializeField]
+		private bool spawnFalling = true;
+
+		[Header("SpriteHandlers")]
+		[SerializeField]
+		private SpriteHandler baseSpriteHandler = default;
+		[SerializeField]
+		private SpriteHandler doorSpriteHandler = default;
+		[SerializeField]
+		private SpriteHandler landingSpriteHandler = default;
+
+		private RegisterObject registerObject;
+		private ClosetControl closetControl;
+
+		[SyncVar(hook = nameof(SyncIsLanding))]
+		private bool isLanding = false;
+
+		private bool IsServer => CustomNetworkManager.IsServer;
+
+		private Vector3Int worldPosition;
+		private Vector3Int WorldPosition
+		{
+			get
+			{
+				if (worldPosition == default)
+				{
+					worldPosition = gameObject.RegisterTile().WorldPosition;
+				}
+				return worldPosition;
+			}
+		}
+
+		private enum BaseSprite
+		{
+			Stationary = 0,
+			Falling = 1
+		}
+
+		#region Lifecycle
+
+		private void Awake()
+		{
+			registerObject = GetComponent<RegisterObject>();
+			closetControl = GetComponent<ClosetControl>();
+		}
+
+		private void Start()
+		{
+			if (spawnFalling)
+			{
+				isLanding = true;
+				SyncIsLanding(isLanding, isLanding);
+			}
+		}
+
+		#endregion Lifecycle
+
+		private void SyncIsLanding(bool oldState, bool newState)
```

Comment:
Codacy found an issue: [Remove this unused method parameter 'oldState'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6316546)